drivers/gpio/gpio_npcx.c

```diff
@@ -0,0 +1,298 @@
+/*
+ * Copyright (c) 2020 Nuvoton Technology Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nuvoton_npcx_gpio
+
+#include <kernel.h>
+#include <device.h>
+#include <drivers/gpio.h>
+#include <soc.h>
+
+#include "gpio_utils.h"
+#include "soc_gpio.h"
+#include "soc_miwu.h"
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(gpio_npcx, LOG_LEVEL_ERR);
+
+/* GPIO module instances declarations */
+static struct device *gpio_devs[];
+static int gpio_devs_count;
+
+/* Driver config */
+struct gpio_npcx_config {
+	/* gpio_driver_config needs to be first */
+	struct gpio_driver_config common;
+	/* GPIO controller base address */
+	uintptr_t base;
+	/* IO port */
+	int port;
+	/* Size of wui mapping array */
+	int wui_size;
+	/* Mapping table between gpio bits and wui */
+	struct npcx_wui wui_maps[];
+};
+
+/* Driver data */
+struct gpio_npcx_data {
+	/* gpio_driver_data needs to be first */
+	struct gpio_driver_data common;
+};
+
+/* Driver convenience defines */
+#define DRV_CONFIG(dev) ((const struct gpio_npcx_config *)(dev)->config)
+
+#define DRV_DATA(dev) ((struct gpio_npcx_data *)(dev)->data)
+
+#define HAL_INSTANCE(dev) (struct gpio_reg *)(DRV_CONFIG(dev)->base)
+
+
+/* Soc specific GPIO functions */
+struct device *soc_get_gpio_dev(int port)
+{
+	if (port >= gpio_devs_count)
+		return NULL;
+
+	return gpio_devs[port];
+}
+
+/* GPIO api functions */
+static int gpio_npcx_config(struct device *dev,
+			     gpio_pin_t pin, gpio_flags_t flags)
+{
+	struct gpio_reg *inst = HAL_INSTANCE(dev);
+	uint32_t mask = BIT(pin);
+
+	/* Don't support simultaneous in/out mode */
+	if (((flags & GPIO_INPUT) != 0) && ((flags & GPIO_OUTPUT) != 0)) {
+		return -ENOTSUP;
+	}
+
+	/* Don't support "open source" mode */
+	if (((flags & GPIO_SINGLE_ENDED) != 0) &&
+	    ((flags & GPIO_LINE_OPEN_DRAIN) == 0)) {
+		return -ENOTSUP;
+	}
+
+	/*
+	 * Configure pin as input, if requested. Output is configured only
+	 * after setting all other attributes, so as not to create a
+	 * temporary incorrect logic state 0:input 1:output
+	 */
+	if (!(flags & GPIO_OUTPUT))
```

Comment:
Can you change these flags to command against `!= 0` as well in this function to remain consistent.

Reply:
@jettr It's hard for the release team to know whether to merge a PR when you simultaneously approve it and ask for changes in a comment.

Reply:
Done. I also modified IS_BIT_SET() macro function to fit MISRA coding guideline in the last commit.

Reply:
@MaureenHelm thanks for the feedback. I was using the process of giving an approval with the knowledge that the author will perform the small changes before submitting code. I forget that authors don't land the changes once they have approval. (I only do this when I also trust the author to do it correctly as well). I am still getting used to the GitHub workflow.

I will wait in the future for all changes to be made (even if small) before marking the approval. Thanks!

Reply:
No worries, thanks